cmd/common.go

```diff
@@ -1,22 +1,158 @@
+	}
 	locations := make([]model.Locations, len(fromArr))
 	for i, v := range fromArr {
 		locations[i] = model.Locations{
 			Magic: strings.TrimSpace(v),
 		}
 	}
-	return locations
+	return locations, false, nil
 }
 
-func inProgressUpdates(ci bool) bool {
-	return !(ci)
+// Maps a location to a measurement ID if possible
+func mapToMeasurementID(location string) (string, error) {
+	if location == "" {
+		return "", nil
+	}
+	if location[0] == '@' {
+		index, err := strconv.Atoi(location[1:])
+		if err != nil {
+			return "", ErrInvalidIndex
+		}
+		return getMeasurementID(index)
+	}
+	if location == "first" {
+		return getMeasurementID(1)
+	}
+	if location == "last" || location == "previous" {
+		return getMeasurementID(-1)
+	}
+	return "", nil
+}
+
+// Returns the measurement ID at the given index from the session history
+func getMeasurementID(index int) (string, error) {
+	if index == 0 {
+		return "", ErrInvalidIndex
+	}
+	f, err := os.Open(getMeasurementsPath())
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return "", ErrorNoPreviousMeasurements
+		}
+		return "", fmt.Errorf("failed to open previous measurements file: %s", err)
+	}
+	defer f.Close()
+	// Read ids from the end of the file
+	if index < 0 {
+		fStats, err := f.Stat()
+		if err != nil {
+			return "", fmt.Errorf("failed to read previous measurements: %s", err)
+		}
+		if fStats.Size() == 0 {
+			return "", ErrorNoPreviousMeasurements
+		}
+		scanner := backscanner.New(f, int(fStats.Size()-1)) // -1 to skip last newline
+		for {
+			index++
+			b, _, err := scanner.LineBytes()
+			if err != nil {
+				if err == io.EOF {
+					return "", ErrIndexOutOfRange
+				}
+				return "", fmt.Errorf("failed to read previous measurements: %s", err)
+			}
+			if index == 0 {
+				return string(b), nil
+			}
+		}
+	}
+	// Read ids from the beginning of the file
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		index--
+		if index == 0 {
+			return scanner.Text(), nil
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return "", fmt.Errorf("failed to read previous measurements: %s", err)
+	}
+	return "", ErrIndexOutOfRange
+}
+
+// Saves the measurement ID to the session history
+func saveMeasurementID(id string) error {
+	_, err := os.Stat(getSessionPath())
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			err := os.Mkdir(getSessionPath(), 0755)
+			if err != nil {
+				return fmt.Errorf("failed to save measurement ID: %s", err)
+			}
+		} else {
+			return fmt.Errorf("failed to save measurement ID: %s", err)
+		}
+	}
+	f, err := os.OpenFile(getMeasurementsPath(), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to save measurement ID: %s", err)
+	}
+	defer f.Close()
+	_, err = f.WriteString(id + "\n")
+	if err != nil {
+		return fmt.Errorf("failed to save measurement ID: %s", err)
+	}
+	return nil
+}
+
+func getSessionPath() string {
+	return fmt.Sprintf("%s/globalping_%d", os.TempDir(), os.Getppid())
```

Comment:
This may not be sufficient as the files may stay stored for a while, and PIDs get recycled. Can we also include the parent start time, as I suggested?